external_parser/parse_example_external.cc

```diff
@@ -13,6 +15,18 @@ bool parser_options::is_enabled() { return binary; }
 std::unique_ptr<parser>
 parser::get_external_parser(vw *all, const input_options &parsed_options) {
   if (parsed_options.ext_opts->binary) {
+    bool binary_to_json = parsed_options.ext_opts->binary_to_json;
+    if (binary_to_json) {
+      std::string infile_name = all->data_filename;
+      std::string outfile_name = infile_name.substr(
```

Comment:
should probably never happen, but maybe check that the filename doesn't already end in `dsjson` and if it does log a warning?

Reply:
or maybe even abort, because it looks like on line 24 we are trying to delete it, so might end up deleting an input file if someone calls this with a dsjson file by mistake